File: src/main/java/com/suifeng/web/italker/push/bean/db/Apply.java
package com.suifeng.web.italker.push.bean.db;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.GenericGenerator;
import org.hibernate.annotations.UpdateTimestamp;

import javax.persistence.*;
import java.time.LocalDateTime;

/**
 * 申请记录表
 */

@Entity
@Table(name = "TB_APPLY")
public class Apply {

    public static final int TYPE_ADD_USER = 1;  //添加用户
    public static final int TYPE_ADD_GROUP = 2; //加入群

    @Id
    @PrimaryKeyJoinColumn
    @GeneratedValue(generator = "uuid")
    @GenericGenerator(name = "uuid", strategy = "uuid2")
    @Column(updatable = false, nullable = false)
    private String id;

    //对申请进行描述
    @Column(nullable = false)
    private String description;

    //附件，可以附带图片地址或者其他
    @Column(columnDefinition = "TEXT")
    private String attach;

    //当前申请的类型
    @Column(nullable = false)
    private int type;

    //目标id
    //type->TYPE_ADD_USER: User.id
    //type->TYPE_ADD_GROUP: Group.id
    @Column(nullable = false)
    private String targetId;

    @CreationTimestamp
    @Column(nullable = false)
    private LocalDateTime createAt = LocalDateTime.now();

    //定义为更新时间戳
    @UpdateTimestamp
    @Column(nullable = false)
    private LocalDateTime updateAt = LocalDateTime.now();

    //申请人 可为空，是系统信息
    @ManyToOne()
    @JoinColumn(name = "applicantId")
    private User applicant;
    @Column(updatable = false, insertable = false)
    private String applicantId;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getAttach() {
        return attach;
    }

    public void setAttach(String attach) {
        this.attach = attach;
    }

    public int getType() {
        return type;
    }

    public void setType(int type) {
        this.type = type;
    }

    public String getTargetId() {
        return targetId;
    }

    public void setTargetId(String targetId) {
        this.targetId = targetId;
    }

    public LocalDateTime getCreateAt() {
        return createAt;
    }

    public void setCreateAt(LocalDateTime createAt) {
        this.createAt = createAt;
    }

    public LocalDateTime getUpdateAt() {
        return updateAt;
    }

    public void setUpdateAt(LocalDateTime updateAt) {
        this.updateAt = updateAt;
    }

    public User getApplicant() {
        return applicant;
    }

    public void setApplicant(User applicant) {
        this.applicant = applicant;
    }

    public String getApplicantId() {
        return applicantId;
    }

    public void setApplicantId(String applicantId) {
        this.applicantId = applicantId;
    }
}

File: src/main/java/com/suifeng/web/italker/push/bean/db/Apply.kt
package com.suifeng.web.italker.push.bean.db

import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.GenericGenerator
import org.hibernate.annotations.UpdateTimestamp

import javax.persistence.*
import java.time.LocalDateTime

/**
 * 申请记录表
 */

@Entity
@Table(name = "TB_APPLY")
class Apply {

    @Id
    @PrimaryKeyJoinColumn
    @GeneratedValue(generator = "uuid")
    @GenericGenerator(name = "uuid", strategy = "uuid2")
    @Column(updatable = false, nullable = false)
    var id: String? = null

    //对申请进行描述
    @Column(nullable = false)
    var description: String? = null

    //附件，可以附带图片地址或者其他
    @Column(columnDefinition = "TEXT")
    var attach: String? = null

    //当前申请的类型
    @Column(nullable = false)
    var type: Int = 0

    //目标id
    //type->TYPE_ADD_USER: User.id
    //type->TYPE_ADD_GROUP: Group.id
    @Column(nullable = false)
    var targetId: String? = null

    @CreationTimestamp
    @Column(nullable = false)
    var createAt = LocalDateTime.now()

    //定义为更新时间戳
    @UpdateTimestamp
    @Column(nullable = false)
    var updateAt = LocalDateTime.now()

    //申请人 可为空，是系统信息
    @ManyToOne
    @JoinColumn(name = "applicantId")
    var applicant: User? = null
    @Column(updatable = false, insertable = false)
    var applicantId: String? = null

    companion object {

        val TYPE_ADD_USER = 1  //添加用户
        val TYPE_ADD_GROUP = 2 //加入群
    }
}
